### DISCUSSION TITLE
Need to restart Webots for the controller (Python) to work

### DISCUSSION THREAD

**Discussion by thephet** [cyberbotics/webots#5876] [Q&A]
Hi,

I am trying to replicate the Epson Scara T6 example. To do so, I start from an empty project. I add an Epson T6 robot, and I set it as supervisor. I also set a staticBase. I select scara_fruit as the controller. I add a transform node to its handslot, which I def as VACCUM. Then I add the conveyor belt, and I fill the belt with apples and oranges. One apple defined es fruit0, and one orange defined as fruit1.

If I then run the simulation, I receive a Python error. If I close and open again Webots, the simulation works as expected.

This error happens both in Linux and Windows.

Doing some testing around, it seems the point at which the controller is chosen is very important. If you do it before some of the DEFs are set, then it fails. But if you set the controller at the end, it seems to work without needing a restart.

**Comment by omichel** [comment]
This is known bug of Webots R2023a. Upgrading to a recent [nightly build](https://github.com/cyberbotics/webots/releases) should fix it.

**Reply by thephet** [reply]
@omichel Thanks. I use it in windows with a few students. So I guess I wasn't doing anything wrong, and the best thing is just to ask them to restart it.

**Reply by omichel** [reply]
Yes, or install a new version of Webots which might be more comfortable for your students.
